### PR TITLE
Issue 371 configurable parameter

### PR DESCRIPTION
**Pull request by EliFeinberg**
Adds in features to runner and runtime to accommodate global variables defined from the yaml global_module_def

### CODE REVIEW

File: common/data_format.hpp [hunk 1]
```diff
@@ -17,37 +17,46 @@
 #define USE_ALT_EYE_FORMAT
 
 namespace ILLIXR {
-using ullong = unsigned long long;
 
-struct cam_type : switchboard::event {
```
Comment (qinjunj) [inline common/data_format.hpp:22]: Why reverting commit #363 ?

File: common/data_format.hpp [hunk 2]
```diff
@@ -20,23 +20,15 @@ namespace ILLIXR {
 
 using ullong = unsigned long long;
 
-// Data type that combines the IMU and camera data at a certain timestamp.
-// If there is only IMU data for a certain timestamp, img0 and img1 will be null
-// time is the current UNIX time where dataset_time is the time read from the csv
-struct imu_cam_type : public switchboard::event {
```
Comment (qinjunj) [inline common/data_format.hpp]: #363 is a large commit that contains a lot of changes. You may need to rebase your changes on top of the current master, and make sure the `revert #363` is not included.

File: common/error_util.hpp [hunk 3]
```diff
@@ -38,12 +38,12 @@ inline void report_and_clear_errno([[maybe_unused]] const std::string& file, [[m
                                    [[maybe_unused]] const std::string& function, [[maybe_unused]] const std::string& msg = "") {
 #ifndef NDEBUG
     if (errno > 0) {
-        if (ILLIXR::ENABLE_VERBOSE_ERRORS) {
+        // if (ILLIXR::ENABLE_VERBOSE_ERRORS) {
```
Comment (qinjunj) [inline common/error_util.hpp:41]: `ENABLE_VERBOSE_ERRORS` is still defined above. We want to keep this config variable. Please change it according to the new way of defining configs.

File: offload_data/plugin.cpp [hunk 4]
```diff
@@ -43,8 +47,8 @@ class offload_data : public plugin {
 
     virtual ~offload_data() override {
         // Write offloaded data from memory to disk
-        if (enable_offload) {
-            boost::filesystem::path p(obj_dir);
+        if (_m_enable_offload) {
+            boost::filesystem::path p(obj_dir); // maybe replace with _m_obj_dir
```
Comment (qinjunj) [inline offload_data/plugin.cpp:51]: Yes, `OFFLOAD_PATH` is defined in `configs/global_module_defs.yaml` as also `metrics/offloaded_data/`, so it's replaceable.

**Review by qinjunj — COMMENTED** [review]
For constant definitions in `common/const_decls.hpp`, how about we put their default values rather than `dev/null`?

**Comment by astro-friedel** [comment]
This PR will be superceded by upcoming switchboard updates and changes

**Pull request closed without merging**